### PR TITLE
added AuthenticationFilter, AuthenticationService, JwtUtil;

### PR DESCRIPTION
**Pull request by mykolaskrypka**
I do not know why - requestMatchers("/auth/**") do not working

### CODE REVIEW

**Comment by mykolaskrypka** [comment]
Please help, can not find why it still goes through the AuthenticateFilter when i make request to the /api/auth/login endpoint

File: src/main/java/mate/academy/security/AuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    ) throws IOException, ServletException {
+
+        String token = extractToken(request);
+        boolean isTokenValid = jwtUtil.isValidToken(token);
+        if (token != null & isTokenValid) {
+            String username = jwtUtil.getUsername(token);
+            UserDetails userDetails = userDetailsService.loadUserByUsername(username);
+            Authentication authentication = new UsernamePasswordAuthenticationToken(
+                    userDetails, null, userDetails.getAuthorities()
+            );
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/security/AuthenticationFilter.java]:
```suggestion
                    userDetails, null, userDetails.getAuthorities());
```

File: src/main/java/mate/academy/security/AuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/security/AuthenticationFilter.java]:
```suggestion
```

File: src/main/java/mate/academy/security/AuthenticationService.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+@Service
+@RequiredArgsConstructor
+public class AuthenticationService {
+    private final JwtUtil jwtUtil;
+    private final AuthenticationManager authenticationManager;
+
+    public UserLoginResponseDto authenticate(UserLoginRequestDto requestDto) {
+        final Authentication authentication = authenticationManager.authenticate(
+                new UsernamePasswordAuthenticationToken(requestDto.email(), requestDto.password())
+        );
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/security/AuthenticationService.java]: Don`t replace ); for new line, it make no sense and looks not good

File: src/main/java/mate/academy/security/AuthenticationService.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        String token = jwtUtil.generateToken(authentication.getName());
+        return new UserLoginResponseDto(token);
+    }
+
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/security/AuthenticationService.java]:
```suggestion
```

File: src/main/java/mate/academy/security/JwtUtil.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import java.nio.charset.StandardCharsets;
+import java.security.Key;
+import java.util.Date;
+import java.util.function.Function;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class JwtUtil {
+
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/security/JwtUtil.java]:
```suggestion
```

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/security/JwtUtil.java (hunk 5 above)
Comment (mykolaskrypka) [reply src/main/java/mate/academy/security/JwtUtil.java]: @Sanderfuck please, read the comment

File: src/main/java/mate/academy/controller/TestController.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api/test")
+public class TestController {
```
Comment (Rommelua) [inline src/main/java/mate/academy/controller/TestController.java]: delete it

Comment (mykolaskrypka) [reply src/main/java/mate/academy/controller/TestController.java]: @Rommelua need help, why it still goes through the AuthenticateFilter when i make request to the /api/auth/login endpoint

**Pull request merged**